cilium/cmd/config.go

```diff
@@ -70,20 +76,55 @@ func configDaemon(cmd *cobra.Command, opts []string) {
 		}
 	} else if len(opts) == 0 {
 		if command.OutputJSON() {
-			if err := command.PrintOutput(cfgStatus.Realized.Options); err != nil {
-				os.Exit(1)
+			if listReadOnlyConfigurations {
+				if err := command.PrintOutput(cfgStatus.DaemonConfigurationMap); err != nil {
+					Fatalf("Error while printing configmap: %v", err)
+				}
+				return
+			} else {
+				if err := command.PrintOutput(cfgStatus.Realized); err != nil {
+					os.Exit(1)
+				}
+				return
 			}
-			return
 		}
-		dumpConfig(cfgStatus.Immutable)
-		dumpConfig(cfgStatus.Realized.Options)
-		fmt.Printf("%-24s %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
-		fmt.Printf("%-24s %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
-		fmt.Printf("%-24s %s\n", "PolicyEnforcement", cfgStatus.Realized.PolicyEnforcement)
-		if cfgStatus.NodeMonitor != nil {
-			fmt.Printf("%-24s %d\n", "MonitorNumPages", cfgStatus.NodeMonitor.Npages)
+		if listReadOnlyConfigurations {
+			fmt.Print("### Read-Only Configurations ###\n\n")
+			keys := make([]string, 0, len(cfgStatus.DaemonConfigurationMap))
+			for k := range cfgStatus.DaemonConfigurationMap {
+				keys = append(keys, k)
+			}
+			sort.Strings(keys)
+			for _, k := range keys {
+				v := cfgStatus.DaemonConfigurationMap[k]
+				if reflect.ValueOf(v).Kind() == reflect.Map {
+					mapString := make(map[string]string)
+					m, ok := v.(map[string]interface{})
+					if ok {
+						fmt.Println(k)
+						for key, value := range m {
+							mapString[key] = fmt.Sprintf("%v", value)
+						}
+						dumpConfig(mapString, true)
+						continue
+					} else {
+						fmt.Fprintf(os.Stderr, "Error: cannot cast daemon config map to map[string]interface{}\n")
+					}
+				}
+				fmt.Printf("%-34s: %v\n", k, v)
+			}
+			fmt.Printf("%-34s: %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
+			fmt.Printf("%-34s: %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
+			if cfgStatus.NodeMonitor != nil {
+				fmt.Printf("%-34s: %d\n", "MonitorNumPages", cfgStatus.NodeMonitor.Npages)
```

Comment:
Why was this option made read-only? (It's the size of the perf ring buffer we use to send notifications to the agent.)